src/runtime-rs/crates/runtimes/src/manager.rs

```diff
@@ -153,26 +154,23 @@ impl RuntimeHandlerManager {
             toml_config: TomlConfig::default(),
             sender,
         };
-        match sandbox_state.sandbox_type {
-            SandboxTYPE::VIRTCONTAINER => {
-                let sandbox = VirtSandbox::restore(sandbox_args, sandbox_state)
-                    .await
-                    .context("failed to restore the sandbox")?;
-                sandbox
-                    .cleanup(&inner.id)
-                    .await
-                    .context("failed to cleanup the resource")?;
-            }
-            SandboxTYPE::LINUXCONTAINER => {
-                // TODO :support linux container (https://github.com/kata-containers/kata-containers/issues/4905)
-                return Ok(());
-            }
-            SandboxTYPE::WASMCONTAINER => {
-                // TODO :support wasm container (https://github.com/kata-containers/kata-containers/issues/4906)
-                return Ok(());
-            }
-        }
 
+        #[cfg(feature = "linux")]
+        // TODO :support linux container (https://github.com/kata-containers/kata-containers/issues/4905)
+        {}
+        #[cfg(feature = "wasm")]
+        // TODO :support wasm container (https://github.com/kata-containers/kata-containers/issues/4906)
+        {}
+        #[cfg(feature = "virt")]
+        {
+            let sandbox = VirtSandbox::restore(sandbox_args, sandbox_state)
+                .await
+                .context("failed to restore the sandbox")?;
+            sandbox
+                .cleanup(&inner.id)
+                .await
+                .context("failed to cleanup the resource")?;
+        }
```

Comment:
What if we enable all `linux`, `wasm` and `virt` features, and let upper layer decide which type of sandbox to create?

Reply:
+1

Reply:
fixed